Core/src/org/sleuthkit/autopsy/logicalimager/dsp/AddLogicalImageTask.java

```diff
@@ -182,70 +274,171 @@ private String addReport(Path reportPath, String reportName) {
+                        filename.replace("'", "''"), parentPath.replace("'", "''"));
+                }
+
+                // TODO - findAllFilesWhere should SQL-escape the query
                 List<AbstractFile> matchedFiles = Case.getCurrentCase().getSleuthkitCase().findAllFilesWhere(query);
                 for (AbstractFile file : matchedFiles) {
-                    addInterestingFile(file, ruleSetName, ruleName);
+                    addInterestingFileToArtifacts(file, ruleSetName, ruleName, artifacts);
                 }
-                lineNumber++;   
+                lineNumber++;
+            } // end reading file
+
+            try {
+                // index the artifact for keyword search
+                blackboard.postArtifacts(artifacts, MODULE_NAME);
+            } catch (Blackboard.BlackboardException ex) {
+                LOGGER.log(Level.SEVERE, "Unable to post artifacts to blackboard", ex); //NON-NLS
             }
         }
-        IngestServices.getInstance().fireModuleDataEvent(new ModuleDataEvent(MODULE_NAME,
-                BlackboardArtifact.ARTIFACT_TYPE.TSK_INTERESTING_FILE_HIT));
     }
 
-    private void addInterestingFile(AbstractFile file, String ruleSetName, String ruleName) throws TskCoreException {
+    private void addInterestingFileToArtifacts(AbstractFile file, String ruleSetName, String ruleName, List<BlackboardArtifact> artifacts) throws TskCoreException {
         Collection<BlackboardAttribute> attributes = new ArrayList<>();
         BlackboardAttribute setNameAttribute = new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_SET_NAME, MODULE_NAME, ruleSetName);
         attributes.add(setNameAttribute);
         BlackboardAttribute ruleNameAttribute = new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_CATEGORY, MODULE_NAME, ruleName);
         attributes.add(ruleNameAttribute);
-        org.sleuthkit.datamodel.Blackboard tskBlackboard = Case.getCurrentCase().getSleuthkitCase().getBlackboard();
-        if (!tskBlackboard.artifactExists(file, BlackboardArtifact.ARTIFACT_TYPE.TSK_INTERESTING_FILE_HIT, attributes)) {
+        if (!blackboard.artifactExists(file, BlackboardArtifact.ARTIFACT_TYPE.TSK_INTERESTING_FILE_HIT, attributes)) {
             BlackboardArtifact artifact = file.newArtifact(BlackboardArtifact.ARTIFACT_TYPE.TSK_INTERESTING_FILE_HIT);
             artifact.addAttributes(attributes);
-            try {
-                // index the artifact for keyword search
-                blackboard.indexArtifact(artifact);
-            } catch (Blackboard.BlackboardException ex) {
-                LOGGER.log(Level.SEVERE, "Unable to index blackboard artifact " + artifact.getArtifactID(), ex); //NON-NLS
+            artifacts.add(artifact);
+        }
+    }
+
+    private void addExtractedFiles(File src, Path resultsPath, List<Content> newDataSources) throws TskCoreException, UnsupportedEncodingException, IOException {
```

Comment:
This doesn't appear to be cancellable?

Reply:
Do you want to rollback the transaction upon cancel?

Reply:
Yes